tools/clang/test/CodeGenSPIRV/method.rwtexture.load.hlsl

```diff
@@ -43,6 +44,10 @@ void main() {
 // CHECK-NEXT: OpStore %e [[r5]]
   int3 e   = int3buf.Load(0);
 
+// CHECK:      [[img6:%\d+]] = OpLoad %type_2d_image_0 %vec1buf
+// CHECK-NEXT:      {{%\d+}} = OpImageRead %v4float [[img6]] {{%\d+}} None
```

Comment:
I checked [rwtexture2d-load](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/rwtexture2d-load), but it's unclear to me why the result type must be `%v4float`.
Could you please explain it to me?

Reply:
We have a comment in the code that says
```
 OpImageFetch and OpImageRead can only fetch a vector of 4 elements.
```

I do see this requirement in the description of OpImageFetch (`Result Type must be a vector of four components`), but I don't see it for OpImageRead. Regardless, I believe it is legal to perform a wider load, and then extract as needed. We always perform OpImageRead / OpImageFetch of a vec4, and then extract the number of elements that we want.

Reply:
This is also the case for GLSL (in GLSL, regardless of the number of elements, the OpImageRead is performed at vec4). Example: http://shader-playground.timjones.io/aa1174cbde94b2b46b71778ad13795d1